src/pages/otp/otp.ts

```diff
@@ -59,8 +64,10 @@ export class OtpPage {
       this.auth.otp(this.unlock.value, true)
         .subscribe((res: any) => {
           this.loading = false;
-          if (res === "ok") {
+          if (res === "ok" && !this.transferToken) {
```

Comment:
Suggestion: refactor this to if (res === "ok") { if (!this.transfer) { ... } else { ... }}